packages/iris-grid/src/IrisGridTableModel.js

```diff
@@ -458,8 +458,36 @@ class IrisGridTableModel extends IrisGridModel {
     return '';
   }
 
+  getColumnsWithLayoutHints = memoize((columns, hints) => {
+    if (hints) {
+      const frontColumns = new Array(hints.frontColumns.length);
+      const backColumns = new Array(hints.backColumns.length);
+      const middleColumns = [];
+
+      columns.forEach(col => {
+        if (hints.frontColumns.includes(col.name)) {
+          frontColumns[hints.frontColumns.indexOf(col.name)] = col;
+        } else if (hints.backColumns.includes(col.name)) {
+          backColumns[hints.backColumns.indexOf(col.name)] = col;
+        } else {
+          middleColumns.push(col);
+        }
+      });
+
+      return [...frontColumns, ...middleColumns, ...backColumns];
+    }
+    return columns;
+  });
+
   get columns() {
-    return this.table.columns;
+    return this.getColumnsWithLayoutHints(
```

Comment:
Before checking this in, we're going to need to update a few other problematic references to `table.columns` directly, and instead have them go through `model.columns`. Otherwise, we may get some weird things out of sync.

The only problem is that I don't know how to get an exhaustive list of those cases easily. You can look for `table.columns`, but it could also be written `const { columns } = table;`. And you'll need to replace those with `model.columns` instead, which could require passing the model through instead of just the table.

IrisGridUtils with all it's dehydrate/hydrate functions has a few places that need to be updated, for example.